tests/fields/test_resolvers.py

```diff
@@ -134,9 +132,11 @@ def adios(self):
 def test_raises_error_when_argument_annotation_missing():
     with pytest.raises(MissingArgumentsAnnotationsError) as e:
 
-        @strawberry.field
-        def hello(self, query) -> str:
-            return "I'm a resolver"
+        @strawberry.type
+        class ShouldRaiseHere:
+            @strawberry.field
+            def hello(self, query) -> str:
+                return "I'm a resolver"
```

Comment:
Not sure about this change. Is it no longer a requirement that a field is "complete" until it's added to a type?

Reply:
There are some stuff that cannot be determined until the origin is determined, for basic fields we must kniw the python name and the origin so fields are not evaluated at all until they are added in a type. I cant see a use case where this might cause problems, do you?